zk-chat-server-lib/src/semaphore/index.ts

```diff
@@ -35,63 +35,78 @@ class SemaphoreSynchronizer {
      * Sync commitments on startup, and schedule a job to sync on a regular interval.
      */
     public sync = async() => {
+        console.log("!@# src/semaphore/index.ts::sync: this.config");
         await this.syncCommitmentsFromSemaphore();
         await this.continuousSync();
     }
 
     public syncCommitmentsFromSemaphore = async() => {
         // On startup
-        // 1. Get all groups from interrep
-        const allGroupsOnNet: IInterRepGroupV2[] = await semaphoreFunctions.getAllGroups(this.config.interepUrl);
+        // 1. Get all groups from Semaphore
+        const allGroupsOnNet: ISemaphoreRepGroupV2[] = await semaphoreFunctions.getAllGroups(this.config.interepUrl);
         const groupsInDb: IGroup[] = await this.groupService.getGroups();
 
         let tree_root_changed = false;
 
         // 2. For each group, check the status in database. Only load new members for group if the size in db is different than the new size of the group
         for (let g of allGroupsOnNet) {
-            const g_id = g.provider + "_" + g.name;
-            const groupInDb: IGroup | undefined = groupsInDb.find(x => x.group_id == g_id && x.name == g.name && x.provider == g.provider);
-
+            const groupMembers = await this.loadGroupMembers(g.id);
```

Comment:
Would it be problematic if there were lots of group members?

Reply:
In theory pagination might help with it but because of the amount of Zuzalu participants I thought it might be better to do it directly rather than with pagination. "Trying to make it cleaner"

Reply:
Sounds reasonable to me:)